packages/manager/apps/pci-databases-analytics/src/pages/services/[serviceId]/indexPatterns/_components/PatternsTableColumns.component.tsx

```diff
@@ -0,0 +1,101 @@
+import { useTranslation } from 'react-i18next';
+import { ColumnDef } from '@tanstack/react-table';
+
+import { MoreHorizontal } from 'lucide-react';
+import {
+  Tooltip,
+  TooltipProvider,
+  TooltipTrigger,
+} from '@/components/ui/tooltip';
+import {
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuItem,
+  DropdownMenuTrigger,
+} from '@/components/ui/dropdown-menu';
+import { Button } from '@/components/ui/button';
+
+import * as database from '@/types/cloud/project/database';
+import { useServiceData } from '../../Service.context';
+import { DatatableSortableHeader } from '@/components/data-table/DatatableSortableHeader.component';
+
+interface PatternsTableColumnsProps {
+  onDeleteClick: (pattern: database.opensearch.Pattern) => void;
+}
+export const getPatternsColumns = ({
```

Comment:
Same here, it violates the rule of hooks

Reply:
It does not violates the rules of hooks anymore, but I'm a bit confused with the naming as `getPatternsColumns` is more a plain function that return an array of elements than a React Component itself.